auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -399,7 +400,23 @@ public String getProviderId() {
         @Deprecated
         @NonNull
         public List<String> getScopes() {
-            return Collections.emptyList();
+            List<String> permissions;
+            if (mProviderId.equals(GoogleAuthProvider.PROVIDER_ID)) {
+                Scope[] array = ((GoogleSignInOptions)
+                        mParams.getParcelable(ExtraConstants.EXTRA_GOOGLE_SIGN_IN_OPTIONS))
+                        .getScopeArray();
+
+                List<String> scopes = new ArrayList<>();
+                for (Scope scope : array) {
+                    scopes.add(scope.toString());
+                }
+                permissions = scopes;
+            } else if (mProviderId.equals(FacebookAuthProvider.PROVIDER_ID)) {
+                permissions = mParams.getStringArrayList(ExtraConstants.EXTRA_FACEBOOK_PERMISSIONS);
+            } else {
+                permissions = null;
+            }
+            return permissions == null ? Collections.<String>emptyList() : permissions;
```

Comment:
nit: why not just start out by making `permissions` the empty list?

Reply:
I wish, but the Facebook permissions have always been nullable (so I kept it that way) and this null check at the end ensures we catch all the cases.